src/components/prompt/browserprompt.js

```diff
@@ -0,0 +1,104 @@
+define(['dialogHelper', 'layoutManager', 'scrollHelper', 'globalize', 'dom', 'require', 'material-icons', 'emby-button', 'paper-icon-button-light', 'emby-input', 'formDialogStyle'], function (dialogHelper, layoutManager, scrollHelper, globalize, dom, require) {
+    'use strict';
+
+    function setInputProperties(dlg, options) {
+        var txtInput = dlg.querySelector('#txtInput');
+
+        if (txtInput.label) {
+            txtInput.label(options.label || '');
+        } else {
+            txtInput.setAttribute('label', options.label || '');
+        }
+        txtInput.value = options.value || '';
+    }
+
+    function showDialog(options, template) {
+
+        var dialogOptions = {
+            removeOnClose: true,
+            scrollY: false
+        };
+
+        if (layoutManager.tv) {
+            dialogOptions.size = 'fullscreen';
+        }
+
+        var dlg = dialogHelper.createDialog(dialogOptions);
+
+        dlg.classList.add('formDialog');
+
+        dlg.innerHTML = globalize.translateHtml(template, 'core');
+
+        if (layoutManager.tv) {
+            scrollHelper.centerFocus.on(dlg.querySelector('.formDialogContent'), false);
+        } else {
+            dlg.querySelector('.dialogContentInner').classList.add('dialogContentInner-mini');
+            dlg.classList.add('dialog-fullscreen-lowres');
+        }
+
+        dlg.querySelector('.btnCancel').addEventListener('click', function (e) {
+            dialogHelper.close(dlg);
+        });
+
+        dlg.querySelector('.formDialogHeaderTitle').innerHTML = options.title || '';
+
+        if (options.description) {
+            dlg.querySelector('.fieldDescription').innerHTML = options.description;
+        } else {
+            dlg.querySelector('.fieldDescription').classList.add('hide');
+        }
+
+        setInputProperties(dlg, options);
+
+        var submitValue;
+
+        dlg.querySelector('form').addEventListener('submit', function (e) {
+
+            submitValue = dlg.querySelector('#txtInput').value;
+            e.preventDefault();
+            e.stopPropagation();
+
+            // Important, don't close the dialog until after the form has completed submitting, or it will cause an error in Chrome
+            setTimeout(function () {
+                dialogHelper.close(dlg);
+            }, 300);
```

Comment:
can we use `}, 0);` here?

Reply:
This seems like a hack to make sure the submission process finishes first.

Reply:
But that 300ms is an arbitraty thing, a server could be slower.
I was hoping it was just a technique to make this function execute at next possible timeframe (like normal `setTimeout(foo, 0)` does)